packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -130,13 +130,30 @@ function useHasTextDecorationControl( name, element ) {
 	return supports.includes( 'textDecoration' );
 }
 
+function TypographyToolsPanel( { ...props } ) {
+	return <ToolsPanel label={ __( 'Typography' ) } { ...props } />;
+}
+
+const DEFAULT_CONTROLS = {
+	fontFamily: true,
+	fontSize: true,
+	fontAppearance: true,
+	lineHeight: true,
+	letterSpacing: true,
+	textTransform: true,
+	textDecoration: true,
+};
+
 export default function TypographyPanel( {
+	as: Wrapper = TypographyToolsPanel,
 	name,
 	element,
 	value,
 	onChange,
-	inherit,
+	inherit = value,
 	settings,
+	panelId,
```

Comment:
"panelId" and "as" prop go hand in hand. I believe there's so magic in the "InspectorControls" panel that "forces" us to do this. There might be a simpler way to do this but this is fine enough I believe.

Reply:
There's some explanatory docs here:

https://github.com/WordPress/gutenberg/blob/7cfbdfdff1b94ce9794ed738db55c0e38040456d/packages/components/src/tools-panel/types.ts#L24-L28

I don't really understand why it's necessary. Surely each block would have a different `ToolsPanelContext` by virtue of the `key={ clientId }` in `BlockSupportToolsPanel`. If not then that seems like a bug in `ToolsPanel`.

If it truly turns out to be necessary then could make this "more correct" by having an `as` prop for the `ToolsPanelItem`s.

🤷‍♂️ agree it's fine for now.